src/test/java/io/jenkins/plugins/jfrog/JfStepTest.java

```diff
@@ -37,4 +50,91 @@ private static Stream<Arguments> jfrogCLIPathProvider() {
                 Arguments.of(new EnvVars(), true, "jf.exe")
         );
     }
+
+    @Test
+    void getJfrogCliVersionTest() throws IOException, InterruptedException {
+        // Mock the Launcher
+        Launcher launcher = mock(Launcher.class);
+        // Mock the Launcher.ProcStarter
+        Launcher.ProcStarter procStarter = mock(Launcher.ProcStarter.class);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        // Mocks the return value of --version command
+        outputStream.write("jf version 2.31.0 ".getBytes());
+        // Mock the behavior of the Launcher and ProcStarter
+        when(launcher.launch()).thenReturn(procStarter);
+        when(procStarter.cmds(any(ArgumentListBuilder.class))).thenReturn(procStarter);
+        when(procStarter.pwd((FilePath) any())).thenReturn(procStarter);
+        when(procStarter.stdout(any(ByteArrayOutputStream.class))).thenAnswer(invocation -> {
+            ByteArrayOutputStream out = invocation.getArgument(0);
+            out.write(outputStream.toByteArray());
+            return procStarter;
+        });
+        when(procStarter.join()).thenReturn(0);
+
+        // Create an instance of JfStep and call the method
+        JfStep jfStep = new JfStep("--version");
+        JfStep.isWindows = System.getProperty("os.name").toLowerCase().contains("win");
+        Version version = jfStep.getJfrogCliVersion(procStarter);
+
+        // Verify the result
+        assertEquals("2.31.0", version.toString());
+    }
+
+    /**
+     * Tests the addCredentialsArguments method logic with password-stdin vs.-- password flag.
+     * Password-stdin flag should only be set if the CLI version is supported
+     * AND the launcher is not the plugin launcher.
+     * Plugin launchers do not support password-stdin, as they do not have access to the standard input by default.
+     *
+     * @param cliVersion       The CLI version
+     * @param isPluginLauncher Whether the launcher is the plugin launcher
+     * @param expectedOutput   The expected output
+     * @throws IOException error
```

Comment:
Please either remove this line or provide an explanation of the conditions under which an error might be thrown